src/plugins/proxy/src/plugin.cpp

```diff
@@ -0,0 +1,572 @@
+        std::lock_guard<std::mutex> lock(m_plugin_mutex);
+        it = properties.find(ov::device::priorities.name());
+        if (it != properties.end()) {
+            m_configs[config_name][ov::device::priorities.name()] = it->second;
+            // Main device is needed in case if we don't have alias and would like to be able change fallback order per
+            // device
+            if (m_alias_for.empty() && config_name.empty())
+                m_alias_for.insert(it->second.as<std::vector<std::string>>()[0]);
+        }
+    }
+    const std::string primary_dev = get_primary_device(get_device_from_config(properties));
+    auto hw_config = properties;
+    // Add fallback priority to detect supported devices in case of HETERO fallback
+    auto device_priority = get_internal_property(ov::device::priorities.name(), config_name);
+    if (!device_priority.empty())
+        hw_config[ov::device::priorities.name()] = device_priority;
+    auto dev_properties = remove_proxy_properties(hw_config, true);
+    std::string dev_prop_name;
+    ov::DeviceIDParser pr_parser(primary_dev);
+    for (const auto& it : dev_properties) {
+        ov::DeviceIDParser parser(it.first);
+        if (parser.get_device_name() == pr_parser.get_device_name()) {
+            // Add primary device properties to primary device
+            OPENVINO_ASSERT(it.second.is<ov::AnyMap>());
+            auto dev_map = it.second.as<ov::AnyMap>();
+            for (const auto& m_it : dev_map) {
+                // Plugin shouldn't contain the different property for the same key
+                OPENVINO_ASSERT(hw_config.find(m_it.first) == hw_config.end() ||
+                                hw_config.at(m_it.first) == m_it.second);
+                hw_config[m_it.first] = m_it.second;
+            }
+            dev_prop_name = it.first;
+            break;
+        }
+    }
+    {
+        // Cannot change config from different threads
+        std::lock_guard<std::mutex> lock(m_plugin_mutex);
+        for (const auto& it : properties) {
+            // Skip proxy properties
+            if (ov::device::id.name() == it.first || it.first == ov::device::priorities.name() ||
+                it.first == ov::proxy::device_priorities.name() || it.first == ov::proxy::alias_for.name() ||
+                // Skip options from config for primaty device
+                hw_config.find(it.first) != hw_config.end() || (!dev_prop_name.empty() && it.first == dev_prop_name))
+                continue;
+            // Cache proxy and fallback device options
+            m_configs[config_name][it.first] = it.second;
+        }
+    }
+    get_core()->set_property(primary_dev, hw_config);
+}
+
+ov::Any ov::proxy::Plugin::get_property(const std::string& name, const ov::AnyMap& arguments) const {
+    size_t device_id = get_device_from_config(arguments);
+    const std::string config_name = is_device_in_config(arguments) ? std::to_string(device_id) : "";
```

Comment:
Why not "0" dy default?

Reply:
Because "0" is the config for first device, but I would like to return internal property for global device if device wasn't specified